### ISSUE TITLE
Bug: When writing SVG into media bag, getting an "Access violation"

### ISSUE THREAD

**Issue by SommerEngineering** [jgm/pandoc#5436]
**Pandoc version: 2.7.1, Microsoft Windows 10, 64 bit**

I wrote a Lua filter to generate all kinds of figures, cf. https://github.com/pandoc/lua-filters/pull/39

This filter reads...

- PlantUML
- Python e.g. Matplotlib
- Graphviz e.g. dot
- LaTeX e.g. Tikz

... and creates figures with captions.

- For `docx`, `pptx`, and `rtf`, all figures are produced as PNG. There is no issue at all.
- But, the default is `svg` e.g. for producing HTML outputs.

Under Linux, everything works fine (cf. [Travis CI](https://travis-ci.org/pandoc/lua-filters/builds/516833535?utm_source=github_status&utm_medium=notification)). With Windows, I get an access violation:

```
Access violation in generated code when reading 0x9

 Attempting to reconstruct a stack trace...

   Frame                                                                                            Code address
 * 0x57edaf0                                                                                        0x3f8d9d6 c:\users\thors\appdata\local\pandoc\pandoc.exe+0x3b8d9d6
 * 0x57edaf8                                                                                        0x7ffed36300f0 C:\Windows\System32\IMM32.DLL+0xf0
 * 0x57edb00                                                                                        0x57ede60
 * 0x57edb70                                                                                        0x7ffed520baa1 C:\Windows\SYSTEM32\ntdll.dll+0x3baa1
 * 0x57edb78                                                                                        0x79844c1
 * 0x57edb80                                                                                        0x7904720
 * 0x57edb88                                                                                        0x7904700
 * 0x57edb90                                                                                        0x7904630
 * 0x57edb98                                                                                        0xd0
 * 0x57edba0                                                                                        0x7855620
 * 0x57edba8                                                                                        0x79846d1
 * 0x57edbb0                                                                                        0x7984578
 * 0x57edbb8                                                                                        0x78db2c1
 * 0x57edbc0                                                                                        0x4b6fb61 c:\users\thors\appdata\local\pandoc\pandoc.exe+0x476fb61
 * 0x57edbc8                                                                                        0x424e618 c:\users\thors\appdata\local\pandoc\pandoc.exe+0x3e4e618
 * 0x57edbd0                                                                                        0x7894b80
 * 0x57edbd8                                                                                        0x7894b49
 * 0x57edbe0                                                                                        0x4a006ba c:\users\thors\appdata\local\pandoc\pandoc.exe+0x46006ba
 * 0x57edbe8                                                                                        0x4a006ba c:\users\thors\appdata\local\pandoc\pandoc.exe+0x46006ba
 * 0x57edbf0                                                                                        0x424e5f8 c:\users\thors\appdata\local\pandoc\pandoc.exe+0x3e4e5f8
 * 0x57edbf8                                                                                        0x4a006ba c:\users\thors\appdata\local\pandoc\pandoc.exe+0x46006ba
 * 0x57edc00                                                                                        0x7894cc0
 * 0x57edc08                                                                                        0x7891e33
 * 0x57edc10                                                                                        0x782b478
 * 0x57edc18                                                                                        0x782b490
 * 0x57edc20                                                                                        0x782b4a8
 * 0x57edc28                                                                                        0x782b4c0
 * 0x57edc30                                                                                        0x4b6fb61 c:\users\thors\appdata\local\pandoc\pandoc.exe+0x476fb61
 * 0x57edc38                                                                                        0x42407fa c:\users\thors\appdata\local\pandoc\pandoc.exe+0x3e407fa
 * 0x57edc40                                                                                        0x4b6fb61 c:\users\thors\appdata\local\pandoc\pandoc.exe+0x476fb61
 * 0x57edc48                                                                                        0x4a7b859 c:\users\thors\appdata\local\pandoc\pandoc.exe+0x467b859
 * 0x57edc50                                                                                        0x4b6fb61 c:\users\thors\appdata\local\pandoc\pandoc.exe+0x476fb61
 * 0x57edc58                                                                                        0x4b6fb61 c:\users\thors\appdata\local\pandoc\pandoc.exe+0x476fb61
 * 0x57edc60                                                                                        0x41b7a33 c:\users\thors\appdata\local\pandoc\pandoc.exe+0x3db7a33
 * 0x57edc68
```

It happens actually [at line 250](https://github.com/SommerEngineering/lua-filters/blob/32eee8904173736a6ab94c57f889246ce7da2ae2/diagram_generator/diagram_generator.lua#L250) when the SVG from Inkscape (converting a PDF created by LaTeX) gets written into the media bag.

**How to reproduce?**
- Set up Pandoc
- Set up dependencies:
   - PlantUML + Java (`plantuml.jar` accessible by path; `JAVA_HOME` set)
   - Inkscape
   - Python 3 e.g. Anaconda Python
   - Graphviz (`dot.exe` accessible by path)
   - LaTeX (`pdflatex.exe` accessible by path)
- Clone https://github.com/SommerEngineering/lua-filters
- Go to the subfolder `diagram_generator`
- Call Pandoc as `pandoc.exe README.md -f markdown -t html --self-contained --standalone --lua-filter=diagram_generator.lua --metadata=title:"README" --metadata=inkscape_path:"c:\Program Files\Inkscape\inkscape.exe" --metadata=python_path:"c:\ProgramData\Anaconda3\python.exe" --metadata=activate_python_path:"c:\ProgramData\Anaconda3\Scripts\activate.bat" -o README.html`
- The access violation occurs every time (reproducible).

**Edit I:**
I don't know why the problem occurs exactly with LaTeX/Inkscape. The other SVG graphics are processed without any issues. Maybe it's not LaTeX/Inkscape, but it always happens after so and so many SVG graphics.

**Comment by mb21** [comment]
It would be good to have a more minimal test... we've had segfaults on Windows in the past even without using lua-filters, see https://github.com/jgm/pandoc/issues/5037, are the reported cases there working on your system?

**Comment by jgm** [comment]
Please try with the recently released 2.7.2.  (Binaries available under releases.)
It fixes a segfault issue we'd been seeing in Windows.

**Comment by SommerEngineering** [comment]
Great, with Pandoc 2.7.2 everything runs fine.
